pkg/graphdb/graphdb.go

```diff
@@ -378,12 +378,22 @@ func (db *Database) Purge(id string) (int, error) {
 		tx.Rollback()
 		return -1, err
 	}
-
 	changes, err := rows.RowsAffected()
 	if err != nil {
 		return -1, err
 	}
 
+	// Clear who's using this id as parent
+	refs, err := tx.Exec("DELETE FROM edge WHERE parent_id = ?;", id)
```

Comment:
putting on cascade delete on the reference doesn't remove the row in question, I've not digged around so much because i didn't fell super confortable in changing these things in this pr, maybe should be another pr to address correct handling of references using the schema itself. If you look the funcion call above is using the same pattern btw, when if correctly handled by the db only the next function would be needed (the one deleting the row holding the foreign key and triggering the cascade)